ostree/ostree_src.go

```diff
@@ -255,7 +258,12 @@ func (s *ostreeImageSource) readSingleFile(commit, path string) (io.ReadCloser,
 
 // GetBlob returns a stream for the specified blob, and the blob's size.
 func (s *ostreeImageSource) GetBlob(info types.BlobInfo) (io.ReadCloser, int64, error) {
+
 	blob := info.Digest.Hex()
+	compressedBlob, found := s.compressed[info.Digest]
```

Comment:
It’s not in general guaranteed that `LayerInfosForCopy` will be called before `GetBlob`; if `GetBlob` depends on this map, it may need to build it on demand.

Reply:
thanks, I've modified it to check if compressed is initialized, and use `LayerInfosForCopy` when it is `nil`